scripts/pre_commit_linter.py

```diff
@@ -729,10 +729,17 @@ def _pre_commit_linter(all_files):
         process.daemon = False
         process.start()
 
-    for process in linting_processes:
-        # Require timeout parameter to prevent against endless waiting for the
-        # linting function to return.
-        process.join(timeout=200)
+    number_of_files_to_lint = (
+        len(html_files_to_lint_for_css) + len(css_files_to_lint) +
+        len(js_files_to_lint) + len(py_files_to_lint))
+
+    # Require timeout parameter to prevent against endless waiting for the
+    # linting function to return.
+    linting_processes[0].join(
+        timeout=html_files_to_lint_for_css/number_of_files_to_lint)
```

Comment:
Hmm, wouldn't this give a timeout in the range `[0, 1]`? Looks like we'll need a multiplier still?

Reply:
Oh, sorry -- yes, you're right! Shall we make the max time as 600 ms then?

Reply:
Yup, let's just make sure the calculated timeout is always >200 so this actually makes an impact!

Reply:
Ah, we have atm 794 js files to lint out of a total 1640 files. Using 600 ms as the max TIMEOUT_MULTIPLIER gives a timeout of 290 ms, which I think is too close to 200 to make an impact. Shall we increase the TIMEOUT_MULTIPLIER to 1000 ms. What do you suggest?

Reply:
Done, thanks a lot Brian!